bears/c_languages/codeclone_detection/ClangComplexityBear.py

```diff
@@ -0,0 +1,94 @@
+from clang.cindex import Index, CursorKind
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.results.SourceRange import SourceRange
+from bears.c_languages.ClangBear import clang_available
+
+
+class ClangComplexityBear(LocalBear):
+    """
+    Calculates cyclomatic complexity of each function and displays it to the
+    user.
+    """
+    check_prerequisites = classmethod(clang_available)
+    decisive_cursor_kinds = {
+        CursorKind.IF_STMT, CursorKind.WHILE_STMT, CursorKind.FOR_STMT,
+        CursorKind.DEFAULT_STMT, CursorKind.CASE_STMT}
+
+    def function_key_points(self, cursor, top_function_level=False):
+        """
+        Calculates number of function's decision points and exit points.
+
+        :param top_function_level: Whether cursor is in the top level of
+                                   the function.
+        """
+        decisions, exits = 0, 0
+
+        for child in cursor.get_children():
+            if child.kind in self.decisive_cursor_kinds:
+                decisions += 1
+            elif child.kind == CursorKind.RETURN_STMT:
+                exits += 1
+                if top_function_level:
+                    # There is no point to move forward, so just return.
+                    return decisions, exits
+            child_decisions, child_exits = self.function_key_points(child)
+            decisions += child_decisions
+            exits += child_exits
+
+        if top_function_level:
+            # Implicit return statement.
+            exits += 1
+
+        return decisions, exits
+
+    def complexities(self, cursor, filename):
+        """
+        Calculates cyclomatic complexities of functions.
+        """
+
+        file = cursor.location.file
+
+        if file is not None and file.name != filename:
+            # There is nothing to do in another file.
+            return
+
+        if cursor.kind == CursorKind.FUNCTION_DECL:
+            child = next((child for child in cursor.get_children()
+                          if child.kind != CursorKind.PARM_DECL),
+                         None)
```

Comment:
are you taking only the first child on purpose? Using `next` for that is a it weird IMO...